numpy/lib/tests/test__iotools.py

```diff
@@ -198,58 +202,58 @@ def test_upgrademapper(self):
         dateparser = _bytes_to_date
         StringConverter.upgrade_mapper(dateparser, date(2000, 1, 1))
         convert = StringConverter(dateparser, date(2000, 1, 1))
-        test = convert(b'2001-01-01')
+        test = convert('2001-01-01')
         assert_equal(test, date(2001, 1, 1))
-        test = convert(b'2009-01-01')
+        test = convert('2009-01-01')
         assert_equal(test, date(2009, 1, 1))
-        test = convert(b'')
+        test = convert('')
         assert_equal(test, date(2000, 1, 1))
 
     def test_string_to_object(self):
         "Make sure that string-to-object functions are properly recognized"
         conv = StringConverter(_bytes_to_date)
-        assert_equal(conv._mapper[-2][0](0), 0j)
+        assert_equal(conv._mapper[-3][0](0), 0j)
```

Comment:
What happened here? A comment saying what `-3` means would be helpful.

Reply:
It's a list index into a list of tuples. The contents of the list was changed, Self explanatory if you take a look at what `_mapper` is.

Reply:
Well, this test is garbage - `[-3]` is `np.longdouble`, yet it seems to be expecting it to be a complex value.

Reply:
Also `_mapper` is a class property, not an instance one - making this test even more bizarre

Reply:
Yeah, I'm not going to spend a lot of time fixing up crappy old tests/codes as long as they are no worse than they were. Note that the long double converter is also broken.

Reply:
And the bytes converter is never reached.

Reply:
Seems like a fair stance.

Note that my comments below are about new code though!